ui/pages/confirmations/hooks/alerts/useConfirmationOriginAlerts.ts

```diff
@@ -0,0 +1,47 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { TransactionMeta } from '@metamask/transaction-controller';
+
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { Confirmation, SignatureRequestType } from '../../types/confirm';
+import { currentConfirmationSelector } from '../../selectors';
+import { isValidASCIIURL, toPunycodeURL } from '../../utils/confirm';
+import { isSignatureTransactionType } from '../../utils';
+
+const useConfirmationOriginAlerts = (): Alert[] => {
+  const t = useI18nContext();
+
+  const currentConfirmation: Confirmation | undefined = useSelector(
+    currentConfirmationSelector,
+  );
+
+  const origin = isSignatureTransactionType(currentConfirmation)
+    ? (currentConfirmation as SignatureRequestType)?.msgParams?.origin
+    : (currentConfirmation as TransactionMeta)?.origin;
+
+  const isValidOrigin = origin && isValidASCIIURL(origin);
+
+  return useMemo<Alert[]>((): Alert[] => {
+    if (!currentConfirmation || isValidOrigin) {
+      return [];
+    }
+
+    return [
+      {
+        key: 'originSpecialCharacterWarning',
+        reason: t('addressMismatch'),
+        field: 'requestFrom',
```

Comment:
Minor, but we try and use the `RowAlertKey` enum so we don't have to hardcode the same string in the alert and the row.

Reply:
PR is updated 👍